iteration/iteration_test.go

```diff
@@ -38,289 +38,273 @@ func (s *TestIterationRepository) TearDownTest() {
+		require.Nil(t, err)
+		assert.NotEmpty(t, i2.Path)
+		i2.Path.Convert()
+		expectedPath := i2.Path.Convert()
+		require.NotNil(t, i2L)
+		assert.Equal(t, expectedPath, i2L.Path.Convert())
+	})
+
+	t.Run("same iteration name within space", func(t *testing.T) {
+		name := "Iteration name test"
+
+		fxt := tf.NewTestFixture(s.T(), s.DB,
+			tf.Spaces(2),
+			tf.Iterations(1, tf.SetIterationNames(name)),
+		)
+
+		i := *fxt.Iterations[0]
+		space2 := *fxt.Spaces[1]
+
+		// another iteration with same name within same sapce, should fail
+		i2 := i
+		i2.ID = uuid.Nil
+		err := repo.Create(context.Background(), &i2)
+		require.NotNil(t, err)
+		assert.Equal(t, reflect.TypeOf(errors.DataConflictError{}), reflect.TypeOf(err))
+
+		// create iteration with same name in another space, should pass
+		i3 := iteration.Iteration{
+			Name:    name,
+			SpaceID: space2.ID,
+		}
+		err = repo.Create(context.Background(), &i3)
+		require.Nil(t, err)
+		require.NotEqual(t, uuid.Nil, i3.ID)
+	})
 }
 
-func (s *TestIterationRepository) TestListIterationBySpace() {
+func (s *TestIterationRepository) TestLoad() {
 	t := s.T()
 	resource.Require(t, resource.Database)
 
 	repo := iteration.NewIterationRepository(s.DB)
-
-	fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(2))
-
-	for i := 0; i < 3; i++ {
+	t.Run("load root iteration", func(t *testing.T) {
 		start := time.Now()
 		end := start.Add(time.Hour * (24 * 8 * 3))
-		name := "Sprint #2" + strconv.Itoa(i)
+		name := "Sprint #24"
+		name2 := "Sprint #24.1"
+
+		fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(1))
 
 		i := iteration.Iteration{
 			Name:    name,
 			SpaceID: fxt.Spaces[0].ID,
 			StartAt: &start,
 			EndAt:   &end,
 		}
-		e := repo.Create(context.Background(), &i)
-		require.Nil(t, e)
-	}
-	// add iteration to the second space from our fixture
-	e := repo.Create(context.Background(), &iteration.Iteration{
-		Name:    "Other Spring #2",
-		SpaceID: fxt.Spaces[1].ID,
+		repo.Create(context.Background(), &i)
+
+		parentPath := append(i.Path, i.ID)
+		require.NotNil(t, parentPath)
+		i2 := iteration.Iteration{
+			Name:    name2,
+			SpaceID: fxt.Spaces[0].ID,
+			StartAt: &start,
+			EndAt:   &end,
+			Path:    parentPath,
+		}
+		repo.Create(context.Background(), &i2)
+
+		res, err := repo.Root(context.Background(), fxt.Spaces[0].ID)
+		require.Nil(t, err)
+		assert.Equal(t, i.Name, res.Name)
+		assert.Equal(t, i.ID, res.ID)
+		expectedPath := i.Path.Convert()
+		require.NotNil(t, res)
+		assert.Equal(t, expectedPath, res.Path.Convert())
 	})
-	require.Nil(t, e)
 
-	its, err := repo.List(context.Background(), fxt.Spaces[0].ID)
-	assert.Nil(t, err)
-	assert.Len(t, its, 3)
-}
+	t.Run("list by space", func(t *testing.T) {
+		fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(2))
 
-func (s *TestIterationRepository) TestUpdateIteration() {
-	t := s.T()
-	resource.Require(t, resource.Database)
-	repo := iteration.NewIterationRepository(s.DB)
+		for i := 0; i < 3; i++ {
```

Comment:
what about using the TestFixture to create the 3 iterations here ? (AFAICS, you did it in the next test)

Reply:
yes now using test fixture in https://github.com/fabric8-services/fabric8-wit/pull/1719/commits/2949ad3902c40f46bca32e444c41e2c67eb4f458